### PR TITLE
API integration for profile photo upload and delete

### PR DESCRIPTION
**Pull request by DejanKatovic**
### What this PR does (required):
- Develop backend to get profile info, upload profile photo, and delete profile photo
- Complete frontend parts for managing profile photo
- Complete API integration

### Screenshots / Videos (required):
Screenshot from 2021-09-21 05-33-37
Screenshot from 2021-09-21 05-34-10

### CODE REVIEW

File: server/controllers/profile.js [hunk 1]
```diff
@@ -22,5 +40,39 @@ exports.savePhoto = asyncHandler(async (req, res, next) => {
...
   });
 
-  res.status(200).json(urls);
-});
\ No newline at end of file
+  const profile = await Profile.findOneAndUpdate({userId: user.id}, {profilePic: urls[0]});
+  
+  if (profile) {
+    const oldUrl = profile.profilePic;
+    const public_id = oldUrl.substring(oldUrl.lastIndexOf('/') + 1, oldUrl.lastIndexOf('.'));
```
Comment (ezeilo-su) [inline server/controllers/profile.js:47]: Prefer to use `camelCase` for all variable names

File: server/controllers/profile.js [hunk 2]
```diff
@@ -22,5 +40,39 @@ exports.savePhoto = asyncHandler(async (req, res, next) => {
...
+});
+
+exports.deletePhoto = asyncHandler(async (req, res) => {
+  const user = req.user;
+  
+  const profile = await Profile.findOneAndUpdate({userId: user.id}, {profilePic: ''});
+
+  if (profile) {
+    const oldUrl = profile.profilePic;
+    const public_id = oldUrl.substring(oldUrl.lastIndexOf('/') + 1, oldUrl.lastIndexOf('.'));
```
Comment (ezeilo-su) [inline server/controllers/profile.js:68]: Prefer to use `camelCase` for all variable names

File: server/routes/profile.js [hunk 3]
```diff
@@ -2,12 +2,17 @@ const express = require("express");
...
 
 const upload = multer({dest: './uploads'});
 
-// router.route("/photo").post(protect, savePhoto);
+router.route("/get-profile")
+	.get(protect, getProfile);
+
 router.route("/save-photo")
-	.post(upload.fields([{name: 'photos'}]), savePhoto);
+	.post(protect, upload.fields([{name: 'photos'}]), savePhoto);
```
Comment (ezeilo-su) [inline server/routes/profile.js:13]: :+1:

**Review by ezeilo-su — COMMENTED** [review]
You've done a good job here :+1: 
Check out a few comments I left and address them accordingly.
Happy coding!

File: client/src/components/ProfilePhoto/ProfilePhoto.tsx [hunk 4]
```diff
@@ -1,20 +1,80 @@
...
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Modal,
+} from '@material-ui/core';
 
 interface Props {
   loggedInUser: User;
+  profile: any;
```
Comment (rajivtitus) [inline client/src/components/ProfilePhoto/ProfilePhoto.tsx:27]: Please avoid using 'any' as a type, don't we have an interface for Profile? :)

File: client/src/components/ProfilePhoto/ProfilePhoto.tsx [hunk 5]
```diff
@@ -24,23 +84,65 @@ const ProfilePhoto = ({ loggedInUser }: Props): JSX.Element => {
...
+      <Dialog
+        open={checkDelete}
+        onClose={handleDeleteClose}
+        aria-labelledby="delete-dialog-title"
+        aria-describedby="delete-dialog-description"
+      >
+        <DialogTitle id="delete-dialog-title">{'Confirm Delete'}</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="delete-dialog-description">
+            {'Do you really want to delete this profile photo?'}
```
Comment (rajivtitus) [inline client/src/components/ProfilePhoto/ProfilePhoto.tsx:134]: 👍🏽

File: client/src/interface/FetchOptions.ts [hunk 6]
```diff
@@ -3,6 +3,6 @@ export interface FetchOptions {
   headers?: {
     'Content-Type': string;
   };
-  body?: string;
+  body?: any;
```
Comment (rajivtitus) [inline client/src/interface/FetchOptions.ts:6]: Please avoid using 'any' as a type

File: client/src/interface/GetProfileApi.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+export interface GetProfileApiSuccess {
+  profile: any;
```
Comment (rajivtitus) [inline client/src/interface/GetProfileApi.ts:2]: If we have a Profile interface we can use it here as well

File: client/src/pages/ProfileSetting/ProfileSetting.tsx [hunk 8]
```diff
@@ -1,18 +1,37 @@
...
   const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const [profile, setProfile] = useState({});
 
   const { loggedInUser } = useAuth();
 
   const history = useHistory();
 
+  useEffect(() => {
```
Comment (rajivtitus) [inline client/src/pages/ProfileSetting/ProfileSetting.tsx:22]: Note: Just keep in mind that we may not need this once Mayank merges his PR as his branch fetches the profile when the user logs in

File: server/controllers/profile.js [hunk 9]
```diff
@@ -1,13 +1,31 @@
...
 
-exports.savePhoto = asyncHandler(async (req, res, next) => {
+exports.getProfile = asyncHandler(async (req, res) => {
+  const user = req.user;
+
+  const profile = await Profile.findOne({userId: user.id});
+  if (profile) {
+    res.status(200).json({
+      success: {
+        profile: profile
```
[the remark this answers is not in the document]
Comment (rajivtitus) [reply server/controllers/profile.js:13]: @DejanKatovic To use shorthand you just write `profile` 

`profile` is the same as `profile: profile`

File: server/controllers/profile.js [hunk 10]
```diff
@@ -1,15 +1,32 @@
...
+      }
+    });
+  }
+  else {
+    res.status(401);
```
Comment (rajivtitus) [inline server/controllers/profile.js:18]: Won't this be a `404` here? :)

File: server/controllers/profile.js [hunk 11]
```diff
@@ -22,5 +39,37 @@ exports.savePhoto = asyncHandler(async (req, res, next) => {
     fs.unlinkSync(photo.path);
   });
 
-  res.status(200).json(urls);
-});
\ No newline at end of file
+  const profile = await Profile.findOneAndUpdate({userId: user.id}, {profilePic: urls[0]});
+  
+  if (profile) {
+    const oldUrl = profile.profilePic;
```
Comment (rajivtitus) [inline server/controllers/profile.js:45]: Why don't we perform a check to see if a user has a `profilePic` and only then delete it?

**Review by rajivtitus — COMMENTED** [review]
Good work with this! Just some comments for you to address. Thanks

File: server/controllers/profile.js (hunk 11 above)
Comment (rajivtitus) [reply server/controllers/profile.js:181]: @DejanKatovic Please address comment above

**Comment by rajivtitus** [comment]
Hey @DejanKatovic, good work so far, just a comment to address. Please tag me in once you've done so. Thanks!

**Comment by rajivtitus** [comment]
> Hey @DejanKatovic, good work so far, just a comment to address. Please tag me in once you've done so. Thanks!

@DejanKatovic please address the pending comment and tag me in once done. Thanks

File: server/controllers/profile.js [hunk 12]
```diff
@@ -170,17 +175,39 @@ exports.savePhoto = asyncHandler(async (req, res, next) => {
...
-  const addPics = await Profile.findOneAndUpdate(
-    { userId: id },
-    { $push: { galleryPics: urls } }
-  );
-  if (!addPics) {
+  const profile = await Profile.findOneAndUpdate({userId: user.id}, {profilePic: urls[0]});
+  
+  if (profile) {
+    const oldUrl = profile.profilePic;
+    if (oldUri) {
```
Comment (ugur-akin) [inline server/controllers/profile.js:182]:
```suggestion
    if (oldUrl) {
```

Comment (rajivtitus) [reply server/controllers/profile.js:182]: @ugur-akin very good catch! The typo threw me off and I was wondering why profilePic validation isn't being performed  😆

**Review by rajivtitus — APPROVED** [review]
LGTM! Please carefully resolve all of the merge conflicts prior to merging.

**Pull request merged**